NatLibFi/Skosmos
# Add translations for tab-alpha.js and vocab-search.js (#1722)

## Review guidelines

Visually observable changes in the browser are primarily related to parts generated by the following files:

- resource/js/tab-alpha.js
- resource/js/vocab-search.js

**Test the user interface:**

- Go: [our Skosmos root]/yso/fi/
- Type "pork" into the vocabulary search bar and observe the different concept types.
- Type "gdfgdgfgf" into the vocabulary search bar and observe the _No results_ text.
- Click on a letter in the alphabetical list and observe the _Loading more items_ text.

**Note:**

- The _msgs_ variable has been removed from the SKOSMOS object.
- The issue in the alphabetical list's Cypress test was caused by a timeout. For some reason in my development environment data transfer from the triple store to the frontend was so slow that an "amazingly" long wait time was required. The question remains as to how this "issue" should be properly addressed.

## Reasons for creating this PR
Improves frontend translation functionalityies.

## Link to relevant issue(s), if any
#1722 

- Closes #1722

## Description of the changes in this PR
Added similar features as in the previous Vue-related translation implementations.

## Known problems or uncertainties in this PR
What if we explore the ways to implement the translation functionality using JavaScript's promise features. By doing so, translations could be loaded in a lazy manner, meaning that if translations need to be awaited, the translation key would briefly appear to the user (likely some milliseconds), but as soon as the translations are ready, the key would be replaced with the actual translation.

This is a very common way to implement similar functionalities. The current implementation works but it has a weakness: if additional dependencies will be needed later that also need to be awaited before the page loads, they must be added to the same functionality (risk of excessive complexity).

Therefore, using the _async -> await -> response -> .then -> .catch_ structure could be considered.

## Checklist

- [x] phpUnit tests pass locally with my changes
- [x] I have added tests that show that the new code works, or tests are not relevant for this PR (e.g. only HTML/CSS changes)
- [x] The PR doesn't reduce accessibility of the front-end code (e.g. tab focus, scaling to different resolutions, use of `.sr-only` class, color contrast)
- [x] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)
